src/generated/xinput.rs

```diff
@@ -733,45 +730,38 @@ impl InputInfoInfo {
             Some(result) => Ok((result, outer_remaining)),
         }
     }
-    pub fn as_key(&self) -> Option<&InputInfoInfoKey> {
-        match self {
-            InputInfoInfo::Key(value) => Some(value),
-            _ => None,
-        }
-    }
-    pub fn as_num_buttons(&self) -> Option<&u16> {
-        match self {
-            InputInfoInfo::NumButtons(value) => Some(value),
-            _ => None,
-        }
-    }
-    pub fn as_valuator(&self) -> Option<&InputInfoInfoValuator> {
-        match self {
-            InputInfoInfo::Valuator(value) => Some(value),
-            _ => None,
-        }
-    }
 }
 impl Serialize for InputInfoInfo {
     type Bytes = Vec<u8>;
-    fn serialize(&self) -> Vec<u8> {
+    fn serialize(&self) -> Self::Bytes {
+        let mut result = Vec::new();
+        self.serialize_into(&mut result);
+        result
+    }
+    fn serialize_into(&self, bytes: &mut Vec<u8>) {
         match self {
-            InputInfoInfo::Key(value) => value.serialize().to_vec(),
-            InputInfoInfo::NumButtons(value) => value.serialize().to_vec(),
-            InputInfoInfo::Valuator(value) => value.serialize(),
+            InputInfoInfo::Key(key) => key.serialize_into(bytes),
+            InputInfoInfo::NumButtons(num_buttons) => {
+                bytes.reserve(2);
+                num_buttons.serialize_into(bytes);
+            }
+            InputInfoInfo::Valuator(valuator) => valuator.serialize_into(bytes),
         }
     }
-    fn serialize_into(&self, bytes: &mut Vec<u8>) {
+}
+impl InputInfoInfo {
+    #[allow(dead_code)]
+    fn switch_expr(&self) -> u8 {
         match self {
-            InputInfoInfo::Key(value) => value.serialize_into(bytes),
-            InputInfoInfo::NumButtons(value) => value.serialize_into(bytes),
-            InputInfoInfo::Valuator(value) => value.serialize_into(bytes),
+            InputInfoInfo::Key(_) => u8::from(InputClass::Key),
+            InputInfoInfo::NumButtons(_) => u8::from(InputClass::Button),
+            InputInfoInfo::Valuator(_) => u8::from(InputClass::Valuator),
         }
     }
 }
+
 #[derive(Debug, Clone, PartialEq, Eq)]
 pub struct InputInfo {
-    pub class_id: InputClass,
```

Comment:
Is this removed because it can be infered from the `info` field?

Oh, wait, this is the fix for that issue somewhere that you mention that you fixed. Okay, found it. :-)